pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,286 @@
+				Description:  "Ldap setting name",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Flag to enable or disable the ldap setting",
+			},
+			"ldap_url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "Location of the LDAP server in the following format: ldap://myserver:myport/dc=sampledomain,dc=com",
+			},
+			"user_dn_pattern": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "A DN pattern that can be used to log users directly in to LDAP. This pattern is used to create a DN string for 'direct' user authentication where the pattern is relative to the base DN in the LDAP URL. The pattern argument {0} is replaced with the username. This only works if anonymous binding is allowed and a direct user DN can be used, which is not the default case for Active Directory (use User DN search filter instead). Example: uid={0},ou=People",
+			},
+			"auto_create_user": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When set, users are automatically created when using LDAP. Otherwise, users are transient and associated with auto-join groups defined in Artifactory.",
+			},
+			"email_attribute": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateIsEmail,
+				Description:  "An attribute that can be used to map a user's email address to a user created automatically in Artifactory.",
+			},
+			"ldap_poisoning_protection": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Protects against LDAP poisoning by filtering out users exposed to vulnerabilities.",
+			},
+			"allow_user_to_access_profile": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Auto created users will have access to their profile page and will be able to perform actions such as generating an API key.",
+			},
+			"paging_support_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When set, supports paging results for the LDAP server. This feature requires that the LDAP server supports a PagedResultsControl configuration.",
+			},
+			"search_filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "A filter expression used to search for the user DN used in LDAP authentication. This is an LDAP search filter (as defined in 'RFC 2254') with optional arguments. In this case, the username is the only argument, and is denoted by '{0}'. Possible examples are: (uid={0}) - This searches for a username match on the attribute. Authentication to LDAP is performed from the DN found if successful.",
```

Comment:
See if you can find RFC 2254 validation library

Reply:
ok